cmd/storage-rest-client.go

```diff
@@ -398,19 +405,47 @@ func (client *storageRESTClient) RenameData(ctx context.Context, srcVolume, srcP
 	return err
 }
 
-func (client *storageRESTClient) ReadVersion(ctx context.Context, volume, path, versionID string) (fi FileInfo, err error) {
+// where we keep old *Readers
+var readMsgpReaderPool = sync.Pool{New: func() interface{} { return &msgp.Reader{} }}
+
+// mspNewReader returns a *Reader that
+// reads from the provided reader. The
+// reader will be buffered.
+func msgpNewReader(r io.Reader) *msgp.Reader {
+	p := readMsgpReaderPool.Get().(*msgp.Reader)
+	if p.R == nil {
+		p.R = xbufio.NewReaderBuf(r, make([]byte, 64*humanize.KiByte))
```

Comment:
64K is a rather large default buffer.

Reply:
It is to cater for `xl.meta + fileSize` @klauspost in-fact keeping this smaller actually is seen to be slower for reads.